lexint.js

```diff
@@ -0,0 +1,114 @@
+module.exports = {
+  preencode,
+  encode,
+  decode
+}
+
+function preencode (state, num) {
+  if (num < 251) {
+    state.end++
+  } else if (num < 256) {
+    state.end += 2
+  } else if (num < 0x10000) {
+    state.end += 3
+  } else if (num < 0x1000000) {
+    state.end += 4
+  } else if (num < 0x100000000) {
+    state.end += 5
+  } else {
+    state.end++
+    const exp = Math.floor(Math.log(num) / Math.log(2)) - 32
+    preencode(state, exp)
+    state.end += 6
```

Comment:
```suggestion
    state.end += 7
```

Won't `exp` be at most 32?

Reply:
I believe `exp <= 2**32`

Reply:
we can bound it if you would rather, but had ported straight from https://github.com/substack/lexicographic-integer

Reply:
Apologies, I misread your comment

Reply:
ahh ok, so it depends whether we only support encoding up to `MAX_SAFE_INTEGER` or `MAX_VALUE`

Reply:
since substack/lexicographic-integer supports up to `MAX_VALUE` I've reverted these changes, but @kasperisager @mafintosh let me know if and I can adapt the tests accordingly